File: Practicals/Practical-1.py
# Caesar Cipher Implementation
def caesar_cipher_encrypt(text, key):
    """
    Encrypts the input text using Caesar Cipher with the given key.
    
    Parameters:
        text (str): The plaintext to be encrypted.
        key (int): The shift value for encryption.
    
    Returns:
        str: The encrypted ciphertext.
    """
    result = ""
    for char in text:
        if char.isupper():
            # Encrypt uppercase characters
            result += chr((ord(char) - ord('A') + key) % 26 + ord('A'))
        elif char.islower():
            # Encrypt lowercase characters
            result += chr((ord(char) - ord('a') + key) % 26 + ord('a'))
        else:
            # Non-alphabetical characters remain the same
            result += char
    return result


def caesar_cipher_decrypt(text, key):
    """
    Decrypts the input text encrypted with Caesar Cipher using the given key.
    
    Parameters:
        text (str): The ciphertext to be decrypted.
        key (int): The shift value used during encryption.
    
    Returns:
        str: The decrypted plaintext.
    """
    result = ""
    for char in text:
        if char.isupper():
            # Decrypt uppercase characters
            result += chr((ord(char) - ord('A') - key) % 26 + ord('A'))
        elif char.islower():
            # Decrypt lowercase characters
            result += chr((ord(char) - ord('a') - key) % 26 + ord('a'))
        else:
            # Non-alphabetical characters remain the same
            result += char
    return result


# Example usage
if __name__ == "__main__":
    plaintext = "Hello, World!"
    key = 3
    print("Original Text: ", plaintext)
    
    # Encrypt the text
    encrypted_text = caesar_cipher_encrypt(plaintext, key)
    print("Encrypted Text: ", encrypted_text)
    
    # Decrypt the text
    decrypted_text = caesar_cipher_decrypt(encrypted_text, key)
    print("Decrypted Text: ", decrypted_text)

File: Practicals/Practical_1.java
package Practicals;

public class Practical_1 {
     // Method to encrypt the plaintext
     public static String encrypt(String text, int key) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < text.length(); i++) {
            char character = text.charAt(i);

            if (Character.isUpperCase(character)) {
                // Encrypt uppercase characters
                char encryptedChar = (char) (((character - 'A' + key) % 26) + 'A');
                result.append(encryptedChar);
            } else if (Character.isLowerCase(character)) {
                // Encrypt lowercase characters
                char encryptedChar = (char) (((character - 'a' + key) % 26) + 'a');
                result.append(encryptedChar);
            } else {
                // Non-alphabetical characters remain unchanged
                result.append(character);
            }
        }
        return result.toString();
    }

    // Method to decrypt the ciphertext
    public static String decrypt(String text, int key) {
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < text.length(); i++) {
            char character = text.charAt(i);

            if (Character.isUpperCase(character)) {
                // Decrypt uppercase characters
                char decryptedChar = (char) (((character - 'A' - key + 26) % 26) + 'A');
                result.append(decryptedChar);
            } else if (Character.isLowerCase(character)) {
                // Decrypt lowercase characters
                char decryptedChar = (char) (((character - 'a' - key + 26) % 26) + 'a');
                result.append(decryptedChar);
            } else {
                // Non-alphabetical characters remain unchanged
                result.append(character);
            }
        }
        return result.toString();
    }
    public static void main(String[] args) {
        String plaintext = "Hello, World!";
        int key = 3;

        // Encryption
        String encryptedText = encrypt(plaintext, key);
        System.out.println("Original Text: " + plaintext);
        System.out.println("Encrypted Text: " + encryptedText);

        // Decryption
        String decryptedText = decrypt(encryptedText, key);
        System.out.println("Decrypted Text: " + decryptedText);
    }
}
